Knossos.NET/ViewModels/Templates/DebugFiltersViewModel.cs

```diff
@@ -0,0 +1,248 @@
+                            } else if (line.StartsWith("-"))
+                            {
+                                enable = false;
+                                option = line.Substring(1).Trim();
+                            }
+
+                            if (option != string.Empty)
+                            {
+                                bool found = false;
+                                foreach (var category in Categories)
+                                {
+                                    foreach (var filter in category.Filters)
+                                    {
+                                        if (filter.Filter.Trim().ToLower() == option.ToLower())
+                                        {
+                                            filter.Enabled = enable;
+                                            found = true;
+                                            break;
+                                        }
+                                        
+                                    }
+                                    if (found) 
+                                    {
+                                        break;
+                                    }
+                                }
+
+                                // if we didn't have a match in the standard categories
+                                if (!found)
+                                {
+                                    foreach (var category in Categories)
+                                    {
+                                        if (category.Typename == "Custom")
+                                        {
+                                            found = true;
+                                            category.AddCustom(option, enable);
+                                            break;
+                                        }
+                                    }
+                                }
+
+                                // if we also didn't have a custom category setup.
+                                if (!found)
+                                {
+                                    var customCategory = new DebugFilterCategory();
+                                    customCategory.Typename = "Custom";
+                                    customCategory.AddCustom(option, enable); 
+                                    Categories.Insert(0, customCategory);
+                                }
+                            }
+                        }
+                    }
+                }                
+            }
+            catch(Exception ex)
+            {
+                Log.Add(Log.LogSeverity.Warning, "DebugFiltersViewModel.LoadDebugRepo()", ex);
+            }
+
+            Instance = this;
+            Initialized = true;
+        }
+
+        public DebugFiltersViewModel()
+        {
+            LoadDebugRepo();
```

Comment:
This generates a warning:
> Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.

I'm not sure of the proper way to address that here but it would be good thing to fix.

Reply:
I saw that warning, and I'm still getting used to asynchronous programming.   But I'll try to do some research and get that to go away.

Reply:
The change works for me.